src/pages/profile/messaging.ts

```diff
@@ -46,7 +46,8 @@ class ContactListComponent
 
   async onAddContact() {
     if (this.newContact.did) {
-      ContactService.addContact(this.newContact as Contact)
+      if (!ContactService.getContact(this.newContact.did))
+        ContactService.addContact(this.newContact as Contact)
```

Comment:
I'm going to approve and merge this still but I strongly recommend always using brackets for multiline if statements. There's some fun anecdotes of someone coming in later to add more to the if body, add a properly indented line but fail to catch that the added indented line isn't a part of the if body.